driver/ni/analog_read.cpp

```diff
@@ -21,13 +21,28 @@ using json = nlohmann::json;
 
 void ni::AnalogReadSource::parse_channels(config::Parser &parser) {
     std::uint64_t c_count = 0;
+    LOG(INFO) << "channel config:" << parser.get_json().dump(4);
     parser.iter("channels",
                 [&](config::Parser &channel_builder) {
                     ni::ChannelConfig config;
                     // analog channel names are formatted: <device_name>/ai<port>
                     std::string port = std::to_string(
                         channel_builder.required<std::uint64_t>("port"));
-                    std::string name = this->reader_config.device_name;
+
+                    std::string name;
+                    if(this->reader_config.device_key != "cross-device"){
+                        name = this->reader_config.device_name;
+                    } else {
+                        auto device_key = channel_builder.required<std::string>("device");
+                        auto [dev, err] = this->ctx->client->hardware.retrieveDevice(
```

Comment:
@Lham42 At some point we should optimize this to avoid fetching the device for every channel.